src/routes/tasks/components/TaskIOPermissions.tsx

```diff
@@ -0,0 +1,37 @@
+import { Box, Icon } from '@chakra-ui/react';
+import { RiGroupLine } from 'react-icons/ri';
+
+import { PermissionsT } from '@/types/CommonTypes';
+
+import { TaskIOTooltip } from '../TasksUtils';
+
+const TaskIOPermissions = ({
+    permissions,
+}: {
+    permissions?: PermissionsT | null;
+}): JSX.Element => {
+    let label = '';
```

Comment:
The label calculation could be wrapped in a `useMemo` hook. This would ensure that the label is only recalculated when the permissions prop changes, potentially improving performance.

To be tested but it could be something like that:
```
const label = React.useMemo(() => {
        if (!permissions) {
            return 'No permissions available yet';
        }
        if (permissions === null || permissions?.download.public) {
            return 'Accessible by everyone';
        }
        if (permissions.download?.authorized_ids.length) {
            return `Accessible by ${permissions?.download.authorized_ids.join()}`;
        }
        return 'Accessible by owner only';
    }, [permissions]);
```

Reply:
I think the label calculation isn't heavy enough to justify the use of useMemo :)